src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.Windows.cs

```diff
@@ -126,7 +125,7 @@ private unsafe RegistryKey CreateSubKeyInternalCore(string subkey, RegistryKeyPe
 
         private void DeleteSubKeyCore(string subkey, bool throwOnMissingSubKey)
         {
-            int ret = Interop.Advapi32.RegDeleteKeyEx(_hkey, subkey, (int)_regView, 0);
+            int ret = Interop.Advapi32.RegDeleteKeyEx(_hkey!, subkey, (int)_regView, 0);
```

Comment:
I defer to the reviewers who are more familiar with nullability annotations. But it seems a shame to have to sprinkle `!` on just about every use of `_hkey` when as far as I can see it's guaranteed to not be null unless a Dispose has happened. And I assume (but did not check) that every public codepath is protected by `EnsureNotDisposed();` either directly or indirectly. I guess the compiler can't analyze that deeply. @stephentoub is this the cleanest way to do it?

Although I feel like we already had this discussion about Dispose() in a previous PR.

Reply:
If something is initialized in the ctor and only set to null on Dispose, we've generally made the field non-nullable.  If you're violating the contract by using the instance after Dispose, then upholding the contract isn't particularly important.

Reply:
Cool. I see it's also set to null in `Win32Error(..)` but it immediately throws. It's basically a variation of dispose but where the handle is invalid.

Reply:
Awesome.

for those who are interested in the documentation 😉
https://github.com/dotnet/runtime/blob/e041241bd94adca59a6ec90173e1d90599deed87/docs/coding-guidelines/api-guidelines/nullability.md#L53-L54